go/libraries/doltcore/sqle/database.go

```diff
@@ -478,6 +478,25 @@ func (db Database) SetRoot(ctx *sql.Context, newRoot *doltdb.RootValue) error {
 	dsess := DSessFromSess(ctx.Session)
 	dsess.dbRoots[db.name] = dbRoot{hashStr, newRoot}
 
+	_, value := ctx.Get(sql.AutoCommitSessionVar)
```

Comment:
Definitely do not want this logic in here, not even temporarily. Should be handled by a layer above.